pyodide-build/pyodide_build/buildall.py

```diff
@@ -430,7 +431,15 @@ def builder(n):
     )
 
 
-def generate_packages_json(pkg_map: dict[str, BasePackage]) -> dict:
+def generate_package_hash(full_path: Path) -> str:
+    sha256_hash = hashlib.sha256()
+    with open(full_path, "rb") as f:
+        for byte_block in iter(lambda: f.read(4096), b""):
```

Comment:
Maybe the first version in https://stackoverflow.com/a/59056837/1791279 is slightly more readable

Reply:
Agreed - I prefer that too.